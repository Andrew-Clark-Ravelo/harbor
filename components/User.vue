<script setup>
import AxoButton from "./Button.vue";

defineProps({
  user: {
    type: Object,
    required: true,
  },
});

const logout = () => navigateTo("/logout");
</script>

<template>
  <div class="items-center gap-2 p-5 flex flex-col">
    <div class="flex gap-4 items-center mb-4">
      <img :src="user.avatar_url" class="rounded-full w-10 h-10" />
      <p class="m-0">
        Welcome <span class="font-bold">{{ user.login }}</span>
      </p>
    </div>
    <axo-button @click="logout"> Logout </axo-button>
  </div>
</template>
