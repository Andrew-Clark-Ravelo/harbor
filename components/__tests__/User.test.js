import { render } from "@testing-library/vue";
import User from "../User.vue";

test("mounts component", () => {
  const { container } = render(User, {
    props: {
      user: {
        login: "sara",
        avatar_url: "https://test.com",
      },
    },
  });

  expect(container).toMatchSnapshot();
});
