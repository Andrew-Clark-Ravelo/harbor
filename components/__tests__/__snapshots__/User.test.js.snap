// Vitest Snapshot v1

exports[`mounts component 1`] = `
<div>
  <div
    class="items-center gap-2 p-5 flex flex-col"
  >
    <div
      class="flex gap-4 items-center mb-4"
    >
      <img
        class="rounded-full w-10 h-10"
        src="https://test.com"
      />
      <p
        class="m-0"
      >
         Welcome 
        <span
          class="font-bold"
        >
          sara
        </span>
      </p>
    </div>
    <button
      class="text-slate-100 bg-axo-orange hover:bg-axo-orange-dark border-transparent w-40 min-w-max p-3 rounded text-lg pointer transition disabled:opacity-60 disabled:cursor-default border-2"
    >
      
       Logout 
      
    </button>
  </div>
</div>
`;
